### DISCUSSION TITLE
data type of remoteXPC payloads

### DISCUSSION THREAD

**Discussion by Rodionp13** [doronz88/pymobiledevice3#554] [Q&A] [answered]
Good day @doronz88. Talking about xpc_message.py encode/decode functionality: All of the types dicts, arrays, etc. - These are plist data types eventually right ?(Haven't seen any imports related to plist though..) Or these are just raw data which must be identified first: for ex. if this is dict then DICTIONARY=0x0000f000 constant will be added first before the actual dict payload.
As far as I read your docs: requests are http2 with xpc dictionaries as a body content. (Assumed it's a bin plist files)

**Comment by doronz88** [comment, marked as answer]
They are not plist types, but raw xpc structs

**Reply by Rodionp13** [reply]
Yeah, thanks. Sorry for dumb question 😀
